src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -632,14 +709,112 @@ void ompl::geometric::RRTstar::deleteBranch(Motion *motion)
     }
 }
 
-ompl::base::Cost ompl::geometric::RRTstar::costToGo(const Motion *motion, const bool shortest) const
+ompl::base::Cost ompl::geometric::RRTstar::defaultCostToGoHeuristic(const base::State *state, const base::Goal *goal) const
+{
+    if (pdef_->getGoal()->hasType(base::GOAL_STATE) == true)
+    {
+        return opt_->motionCost(state, goal->as<base::GoalState>()->getState());
+    }
+    else if (pdef_->getGoal()->hasType(base::GOAL_REGION) == true)
+    {
+        return base::goalRegionCostToGo(state, goal);
```

Comment:
So, I see now that `GoalState` and `GoalStates` are both derived from `GoalSampleableRegion`, which is itself derived from `GoalRegion`...

I think I need some guidance.

Does that mean `base::goalRegionCostToGo` **is** valid for goals of type `GoalState`? Does it just do the same as what I was doing with the switch? In the case of `GoalStates`, does it find the min over the states or does it assume that the cost-to-go calculated with any one is the same?... I will have to look into it more, but this could really clean up some of my other code. :smile: